pytorch_lightning/utilities/distributed.py

```diff
@@ -345,3 +346,26 @@ def register_ddp_comm_hook(
 
 def tpu_distributed() -> bool:
     return _TPU_AVAILABLE and xm.xrt_world_size() > 1
+
+
+def init_ddp_connection(
+    cluster_environment: ClusterEnvironment,
```

Comment:
the precedence between cluster environment and global_rank/world_size might not be fully clear to callers

Reply:
`global_rank/world_size` is Optional, hence the choice.